src/Features/Core/Portable/Snippets/SnippetChange.cs

```diff
@@ -23,9 +24,16 @@ namespace Microsoft.CodeAnalysis.Snippets
         /// </summary>
         public readonly int? CursorPosition;
 
+        /// <summary>
+        /// The items that we will want to rename as well as the ordering
+        /// in which to visit those items.
+        /// </summary>
+        public readonly List<(string, List<TextSpan>)> Placeholders;
```

Comment:
Is there a better structure to be used here? Used a list to maintain order and use the index as a means of determining the numeric value that should be inserted to convert to an LSP snippet.

Reply:
Plz use ImmutableArray in data types so that there's no chance of mutation.

For a complex type like this also please doc the general shape of this and perhaps have examples of what may be in the individual components.  Given a tuple, naming the members is often useful too (as i really don't know what that `string` or `List<TextSpan>` means) :)